src/style.rs

```diff
@@ -0,0 +1,299 @@
+        separator.max.y = midpoint + self.separator_size * 0.5;
+
+        let response = ui
+            .allocate_rect(separator, Sense::click_and_drag())
+            .on_hover_cursor(CursorIcon::ResizeVertical);
+
+        {
+            let delta = response.drag_delta().y;
+            let range = rect.max.y - rect.min.y;
+
+            let min = (self.separator_extra / range).min(1.0);
+            let max = 1.0 - min;
+            let (min, max) = (min.min(max), max.max(min));
+            *fraction = (*fraction + delta / range).clamp(min, max);
+        }
+
+        let midpoint = rect.min.y + rect.height() * *fraction;
+        separator.min.y = map_to_pixel(
+            midpoint - self.separator_size * 0.5,
+            pixels_per_point,
+            f32::round,
+        );
+        separator.max.y = map_to_pixel(
+            midpoint + self.separator_size * 0.5,
+            pixels_per_point,
+            f32::round,
+        );
+
+        (
+            rect.intersect(Rect::everything_above(separator.min.y)),
+            separator,
+            rect.intersect(Rect::everything_below(separator.max.y)),
+        )
+    }
+
+    pub(crate) fn tab_title(
+        &self,
+        ui: &mut Ui,
+        label: String,
+        active: bool,
+        is_being_dragged: bool,
+    ) -> Response {
+        let px = ui.ctx().pixels_per_point().recip();
+        let rounding = self.tab_rounding;
+
+        let font_id = FontId::proportional(14.0);
+        let galley = ui.painter().layout_no_wrap(label, font_id, self.tab_text);
+
+        let offset = egui::vec2(8.0, 0.0);
+        let text_size = galley.size();
+
+        let mut desired_size = text_size + offset * 2.0;
+        desired_size.y = 24.0;
+
+        let (rect, response) = ui.allocate_at_least(desired_size, Sense::hover());
+        let response = response.on_hover_cursor(CursorIcon::PointingHand);
+
+        match (active, is_being_dragged) {
+            (true, false) => {
+                let mut tab = rect;
+                tab.min.x -= px;
+                tab.max.x += px;
+                ui.painter()
+                    .rect_filled(tab, rounding, self.tab_outline_color);
+
+                tab.min.x += px;
+                tab.max.x -= px;
+                tab.min.y += px;
+                ui.painter()
+                    .rect_filled(tab, rounding, self.background_color);
+            }
+            (true, true) => {
+                let tab = rect;
+
+                ui.painter().rect_stroke(
+                    tab,
+                    self.tab_rounding,
+                    Stroke::new(1.0, self.tab_outline_color),
+                );
+            }
+            _ => (),
+        }
+
+        let pos = Align2::LEFT_TOP
+            .anchor_rect(rect.shrink2(vec2(8.0, 5.0)))
+            .min;
+
+        ui.painter().galley(pos, galley);
+
+        response
+    }
+}
+
+#[derive(Default)]
+pub struct StyleBuilder {
+    style: Style,
+}
+
+impl StyleBuilder {
```

Comment:
Add documentation comments for each public method, briefly explaining exactly what part of the visuals they affect